dev-util/clippy/clippy-9.1.ebuild

```diff
@@ -0,0 +1,58 @@
+# Copyright 2020-2024 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+MY_P="frr-${PV}"
+PYTHON_COMPAT=( python3_{9..12} )
+inherit autotools python-single-r1
+
+DESCRIPTION="Standalone clippy tool built from FRR sources"
+HOMEPAGE="https://frrouting.org/"
+SRC_URI="https://github.com/FRRouting/frr/archive/${MY_P}.tar.gz"
+S="${WORKDIR}/frr-${MY_P}"
+
+LICENSE="GPL-2"
+SLOT="0"
+KEYWORDS="amd64 ~arm64 ~x86"
+REQUIRED_USE="${PYTHON_REQUIRED_USE}"
+
+# standalone clippy does not have any tests
+# restrict to prevent bug 811753
+RESTRICT="test"
+
+DEPEND="
+	${PYTHON_DEPS}
+	virtual/libelf:=
+"
+RDEPEND="${DEPEND}"
+BDEPEND="
+	sys-devel/bison
+	sys-devel/flex
+"
+
+QA_CONFIG_IMPL_DECL_SKIP=(
+	mallinfo	# check in configure is fallback is needed
+	mallinfo2	# check in configure is fallback is needed
+)
+
+src_prepare() {
+	default
+	eautoreconf
+}
+
+src_configure() {
+	# bug #884291
+	unset YACC
+	export LEX=flex
+
+	econf --enable-clippy-only
+}
+
+src_install() {
+	# 830087
+	find "lib" -type f -name "clippy" -print0 |
+		xargs -0 file | grep executable | grep ELF | cut -f 1 -d : |
+		xargs -I '{}' dobin '{}' ||
+		die "Failed to install 'lib/clippy'"
```

Comment:
You're missing a `BDEPEND` on `sys-apps/file`. Also, this is horrible and fragile. Can't you just list specific files instead?

Reply:
This seems to be because clippy isn’t really intended to be a stand-alone package, and it’s a corner case with `dev-build/slibtool`: https://gitweb.gentoo.org/repo/gentoo.git/commit/?id=0e4bdd6bbebca017d6099c62090116306d1c1a89

And I don’t understand why we just don’t integrate clippy with frr (it’s the same tarball anyway), and I don’t any other package depending on it:
```shell
alarig@x280 gentoo % (net-misc/frr =) grep -r dev-util/clippy
net-misc/frr/frr-9.1.ebuild:	~dev-util/clippy-${PV}
net-misc/frr/frr-9.0.2.ebuild:	~dev-util/clippy-${PV}
# metadata and git after that
```

I would say that we could try to remove this ebuild and only use frr, but I don’t really know if I would be someone’s workflow or not.

Reply:
I applied @thesamesam suggestion with `LICENSE="GPL-2+"`

Reply:
```
jkroon@plastiekpoot ~/projects/gentoo (master) $ grep dev-util/clippy */*/*.ebuild
net-misc/frr/frr-9.0.2.ebuild:	~dev-util/clippy-${PV}
```

So at least there are no other references to the clippy *package* even though rust has a clippy USE flag.  The rust USE flag says this "Install clippy, Rust code linter", whereas dev-util/clippy description says "Standalone clippy tool built from FRR sources" - so unless clippy in FRR and rust[clso I'm with @alarig on this one, let's integrate them into a single FRR package and move on with our lives, it would simplify things quite a bit for FRR.

https://github.com/FRRouting/frr/issues/2528 specifically states:

"Clippy is a custom preprocessor for our codebase. It runs as part of the build procedure to generate code for our CLI. You cannot build FRR without Clippy."

clippy is thus a BDEPEND for frr, I'm assuming that frr's make install won't actually install it?  Will this be handled correctly during cross-compile inside of frr or is that perhaps the reason for it being a separate package?

If we need/want to keep the packages separate:

```
jkroon@plastiekpoot /var/tmp/portage/dev-util/clippy-9.0.2/work/frr-frr-9.0.2 $ find "lib" -type f -name "clippy" -print0 | xargs -0 file | grep executable | grep ELF
lib/clippy: ELF 64-bit LSB pie executable, x86-64, version 1 (SYSV), dynamically linked, interpreter /lib64/ld-linux-x86-64.so.2, for GNU/Linux 3.2.0, not stripped
jkroon@plastiekpoot /var/tmp/portage/dev-util/clippy-9.0.2/work/frr-frr-9.0.2 $ 
```

So IMHO we just replace the entire mess with:

```
dobin lib/clippy
```

Reply:
I just merged the two packages